presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2005,6 +2005,27 @@ public PartitionStatistics getStatisticsUpdate()
             return statisticsUpdate;
         }
 
+        public Table getAugmentTableForInTransactionRead()
+        {
+            if (!this.currentLocation.isPresent()) {
+                // partitioned table
+                return this.table;
+            }
+
+            // For unpartitioned table, this method augments the location field of the table
+            // to the staging location.
+            // This way, if the table is accessed in an ongoing transaction, staged data
+            // can be found and accessed.
```

Comment:
Why is this specific to unpartitioned tables?

Reply:
Partitioned tables don't have data stored directly in the table directory.